src/iter/filter_map.rs

```diff
@@ -24,9 +24,9 @@ pub fn new<M, FILTER_OP>(base: M, filter_op: FILTER_OP) -> FilterMap<M, FILTER_O
     }
 }
 
-impl<M, FILTER_OP, R> ParallelIterator for FilterMap<M, FILTER_OP>
+impl<M, P, R> ParallelIterator for FilterMap<M, P>
```

Comment:
I no longer recall why I used `M` for the "base" operations, but it doesn't seem terribly intuitive to me. I see that [the standard library uses `I`](https://doc.rust-lang.org/std/iter/struct.Filter.html), perhaps we should use that (after adopting @cuviper's suggestion to change `ITEM` to `T`)?